### ISSUE TITLE
v:menu.list inconsistent behaviour

### ISSUE THREAD

**Issue by pixelplant** [FluidTYPO3/vhs#1259]
Hello

I'm trying to generate a menu from a list of page ids and so am passing an array to v:menu.list (using TYPO3 v7.6.15 and latest version of vhs)
My code looks like this:
```
<v:menu.list pages="{myPages}" as="myMenu">
    <f:for each="{myMenu}" as="page" iteration="iterator">
        ...blabla...
    </f:for>
</v:menu.list>
```
myPages is a simple PHP array generated in the controller and contains an array with page uids. An f:debug on this array generates this data, for example:
```
array(3 items)
   0 => '147' (3 chars)
   1 => '148' (3 chars)
   2 => '151' (3 chars)
```

Executing this code generates an SQL error as it seems that somewhere in the call the viewhelper loses the list of page ids and sends nothing to the query...
```
SELECT * FROM pages WHERE pid IN () AND pages.deleted=0
```
As you can see, **pid IN ()** is empty.

**On the other hand**, if I try to pass the array directly in the call it works and it fetches the pages.
Ex: this works:
```
<v:menu.list pages="{0: 147, 1: 148, 2: 151}" as="myMenu">
    <f:for each="{myMenu}" as="page" iteration="iterator">
        ...blabla...
    </f:for>
</v:menu.list>
```

**Comment by pixelplant** [comment]
Found the issue. One of the pages sent in the array was hidden (hidden = 1). Still, even with showHiddenInMenu="0" v:menu.list would not skip this page and hence I got the error.
My fix so far was to skip hidden pages when generating this array and then send the proper array to v:menu.list

**Comment by bjo3rnf** [comment]
Hi,

this should be fixed in #1247. Please try with the current development branch.

Cheers
Björn

**Comment by bjo3rnf** [comment]
Closing due to lack of feedback and pointing to #1247
